packages/jsii-diff/bin/jsii-diff.ts

```diff
@@ -43,6 +43,18 @@ async function main(): Promise<number> {
       default: false,
       desc: 'Error on experimental API changes',
     })
+    .option('external-errors', {
+      alias: 'x',
+      type: 'boolean',
+      default: false,
+      desc: 'Error on external API changes',
+    })
+    .option('deprecated-errors', {
```

Comment:
```ts
  /**
   * The API may emit warnings. Backward compatibility is not guaranteed.
   *
   * More information about the deprecation can usually be found in the
   * `deprecated` field.
   */
  Deprecated = 'deprecated',
```

That's not how jsii handles it and currently deprecated features will only emit a warning.
I think jsii models this after Node: https://nodejs.org/api/documentation.html#documentation_stability_index

I can _see_ where you are coming from though. Happy to looking into changing the current behavior in principle.

Reply:
> Is this worth implementing as a bunch of individual flags? Should we do named classes?
> 
> ```
> --errors=stable
> --errors=stable,experimental
> --errors=all
> ```
> 
> ...etc...?

Sure, I was trying to avoid a new major version release. But I guess I can deprecate the previous setting.